### PR TITLE
feat: universal-stash

### PR DESCRIPTION
**Pull request by ChefJerry**
## PR-Codex overview
This PR updates various packages and configurations related to PancakeSwap, including adding new ABIs, SDKs, and improving the universal router setup.

### Detailed summary
- Added `v3PoolAbi` ABI
- Updated SDK versions
- Added new entities and protocols
- Improved universal router setup
- Enhanced error handling in `UserUnexpectedTxError` class

> The following files were skipped due to too many changes: `packages/smart-router/evm/v3-router/utils/index.ts`, `apps/web/src/views/Swap/V3Swap/hooks/index.ts`, `package.json`, `packages/universal-router-sdk/test/utils/permit.ts`, `packages/smart-router/evm/index.test.ts`, `apps/web/src/hooks/usePermit2Allowance.ts`, `packages/universal-router-sdk/src/entities/types.ts`, `packages/position-managers/package.json`, `packages/universal-router-sdk/README.md`, `packages/widgets-internal/swap/SwapTransactionReceiptModalContentV1.tsx`, `packages/permit2-sdk/tsconfig.json`, `packages/universal-router-sdk/tsconfig.json`, `apps/web/src/hooks/useGetENSAddressByName.ts`, `packages/wagmi/package.json`, `apps/web/package.json`, `apps/web/src/hooks/useTokenAllowance.ts`, `apps/web/src/views/Swap/V3Swap/hooks/useSwapCurrency.ts`, `packages/v3-sdk/src/abi/V3PoolAbi.ts`, `packages/permit2-sdk/src/utils/utils.ts`, `apps/web/src/views/Swap/V3Swap/containers/SwapCommitButton.tsx`, `apps/web/src/views/Swap/V3Swap/hooks/useSwapCallback.ts`, `apps/web/src/views/Swap/V3Swap/hooks/useSwapCallArguments.ts`, `apps/web/src/hooks/useCallWithGasPrice.ts`, `apps/web/src/views/Swap/V3Swap/types.ts`, `apps/web/src/views/Swap/V3Swap/hooks/useSwapBestTrade.ts`, `apps/web/src/state/transactions/reducer.ts`, `apps/web/src/views/Swap/V3Swap/containers/MMCommitButton.tsx`, `apps/web/src/hooks/useApproveCallback.ts`, `packages/permit2-sdk/package.json`, `apps/web/src/hooks/usePublicNodeWaitForTransaction.ts`, `packages/widgets-internal/swap/index.ts`, `packages/widgets-internal/swap/TransactionErrorContent.tsx`, `packages/widgets-internal/swap/ApproveModalContentV1.tsx`, `packages/widgets-internal/swap/SwapPendingModalContentV1.tsx`, `packages/universal-router-sdk/test/fixtures/clients.ts`, `packages/widgets-internal/swap/SwapTransactionReceiptModalContent.tsx`, `packages/universal-router-sdk/test/utils/calldataDecode.ts`, `packages/universal-router-sdk/src/entities/protocols/unwrapWETH.ts`, `packages/universal-router-sdk/package.json`, `apps/web/src/hooks/useWritePermit.ts`, `apps/web/src/views/Swap/V3Swap/hooks/useWallchain.ts`, `apps/web/src/utils/wagmi.ts`, `apps/web/src/hooks/usePermit2Details.ts`, `apps/web/src/views/Swap/V3Swap/components/TransactionConfirmSwapContent.tsx`, `apps/web/src/hooks/usePermit2Requires.ts`, `apps/web/src/views/Swap/components/SwapModalHeader.tsx`, `packages/universal-router-sdk/test/fixtures/constants/tokens.ts`, `apps/web/src/views/Swap/V3Swap/containers/MMCommitButtonV2.tsx`, `packages/universal-router-sdk/src/constants.ts`, `apps/web/src/config/wallet.ts`, `packages/widgets-internal/swap/styles.tsx`, `apps/web/src/views/Swap/MMLinkPools/components/MMCommitButton.tsx`, `apps/web/src/views/Swap/V3Swap/hooks/useSendSwapTransaction.ts`, `packages/widgets-internal/swap/SwapPendingModalContent.tsx`, `apps/web/src/views/Swap/V3Swap/containers/ApproveStepFlow.tsx`, `apps/web/src/views/Swap/V3Swap/hooks/useSwapCallArgumentsV2.ts`, `packages/universal-router-sdk/test/fixtures/constants/chains.ts`, `packages/universal-router-sdk/src/abis/ERC20.ts`, `apps/web/src/views/Swap/V3Swap/components/SwapModalFooter.tsx`, `packages/universal-router-sdk/src/swapRouter.ts`, `packages/permit2-sdk/src/constants.ts`, `packages/permit2-sdk/src/allowanceTransfer.ts`, `apps/web/src/views/Swap/MMLinkPools/hooks/useApprove.ts`, `apps/web/src/hooks/usePermit2.ts`, `apps/web/src/state/transactions/hooks.tsx`, `apps/web/src/views/Swap/V3Swap/index.tsx`, `packages/localization/src/config/translations.json`, `packages/universal-router-sdk/src/utils/inputTokens.ts`, `packages/widgets-internal/swap/useUnmountingAnimation.ts`, `apps/web/src/views/Swap/V3Swap/containers/ApproveStepFlowV2.tsx`, `packages/universal-router-sdk/src/abis/NFT20.ts`, `apps/web/src/views/Swap/V3Swap/hooks/useSwapCallbackV2.ts`, `packages/permit2-sdk/src/signatureTransfer.ts`, `packages/widgets-internal/swap/ApproveModalContent.tsx`, `packages/universal-router-sdk/test/utils/buildTrade.ts`, `apps/web/src/views/Swap/MMLinkPools/hooks/useSwapCallArguments.ts`, `apps/web/src/views/Swap/V3Swap/containers/ConfirmSwapModal.tsx`, `packages/universal-router-sdk/test/fixtures/address.ts`, `apps/web/src/views/Swap/MMLinkPools/components/MMCommitButtonV2.tsx`, `packages/widgets-internal/swap/Logos.tsx`, `apps/web/src/views/Swap/MMLinkPools/hooks/useSwapCallback.ts`, `apps/web/src/views/Swap/MMLinkPools/hooks/useMMConfirmModalState.ts`, `apps/web/src/views/Swap/V3Swap/hooks/useConfirmModalState.tsx`, `packages/universal-router-sdk/src/abis/NFTXZap.ts`, `apps/web/src/views/Swap/V3Swap/containers/ConfirmSwapModalV2.tsx`, `packages/universal-router-sdk/src/abis/UniversalRouter.ts`, `packages/universal-router-sdk/src/utils/routerCommands.ts`, `apps/web/src/views/Swap/V3Swap/containers/SwapCommitButtonV2.tsx`, `apps/web/src/views/Swap/V3Swap/hooks/useConfirmModalStateV2.ts`, `packages/permit2-sdk/src/abis/Permit2.ts`, `packages/universal-router-sdk/src/abis/IUniswapV3Pool.ts`, `packages/universal-router-sdk/src/entities/protocols/pancakeswap.ts`, `packages/universal-router-sdk/src/abis/X2Y2.ts`, `packages/universal-router-sdk/src/abis/Sudoswap.ts`, `packages/universal-router-sdk/src/abis/Foundation.ts`, `packages/universal-router-sdk/src/abis/LooksRareV2.ts`, `packages/universal-router-sdk/src/abis/Element.ts`, `packages/universal-router-sdk/src/abis/Seaport.ts`, `pnpm-lock.yaml`, `packages/universal-router-sdk/test/__snapshots__/trade.test.ts.snap`, `packages/universal-router-sdk/test/trade.test.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by chefjackson** [comment]
Could you help cleanup the ts errors?

File: .changeset/moody-games-wash.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@pancakeswap/wagmi': minor
+---
+
+Remove binance extension wallet connector
```
Comment (chef-jojo) [inline .changeset/moody-games-wash.md:5]: @pancakeswap/wagmi: major
@pancakeswap/v3-sdk: patch
@pancakeswap/smart-router: patch
Let's also publich the permit2-sdk and universal router sdk

File: apps/web/src/hooks/usePermit2Requires.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  const { account } = useAccountActiveChain()
+  const { allowance, refetch } = usePermit2Allowance(account, amount?.currency)
+  const { data } = usePermit2Details(account, amount?.currency, spender)
+  const { amount: permitAmount, expiration = 0n } = data ?? {}
+  const now = useCurrentBlockTimestamp() ?? 0n
+
+  const requireRevoke = useMemo((): boolean => {
+    const isMainnetUSDT =
+      amount?.currency?.chainId === ethereumTokens.usdt.chainId &&
+      isAddressEqual(amount?.currency.address, ethereumTokens.usdt.address)
```
Comment (chef-jojo) [inline apps/web/src/hooks/usePermit2Requires.ts:22]: `isAddressEqual` will throw error

Comment (chef-jojo) [reply apps/web/src/hooks/usePermit2Requires.ts:22]: nvm, the address is not optional there, although we have the `?.` syntax

File: apps/web/src/views/Swap/V3Swap/containers/SwapCommitButtonV2.tsx [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+    [Field.OUTPUT]: relevantTokenBalances[1],
+  }
+  const parsedAmounts = useParsedAmounts(trade, currencyBalances, false)
+  const parsedIndependentFieldAmount = parsedAmounts[independentField]
+  const swapInputError = useSwapInputError(trade, currencyBalances)
+  const [tradeToConfirm, setTradeToConfirm] = useState<SmartRouterTrade<TradeType> | undefined>(undefined)
+  const [indirectlyOpenConfirmModalState, setIndirectlyOpenConfirmModalState] = useState(false)
+
+  const { callToAction, confirmState, txHash, confirmActions, errorMessage, resetState } = useConfirmModalStateV2(
+    trade,
```
Comment (chefjackson) [inline apps/web/src/views/Swap/V3Swap/containers/SwapCommitButtonV2.tsx:158]: Should use `tradeToConfirm` insteand of `trade` in non-expert mode?

File: apps/web/src/views/Swap/MMLinkPools/components/MMCommitButtonV2.tsx [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+  const [tradeToConfirm, setTradeToConfirm] = useState<SmartRouterTrade<TradeType> | undefined>(undefined)
+  const amountToApprove = useMemo(() => {
+    return mmTradeInfo?.slippageAdjustedAmounts[Field.INPUT]
+  }, [mmTradeInfo])
+  const mmSpender = useMemo(() => {
+    return mmTradeInfo?.routerAddress as Address | undefined
+  }, [mmTradeInfo])
+  const swapCalls = useSwapCallArguments(rfqTrade.trade, rfqTrade.rfq ?? undefined, recipient ?? undefined)
+  const { resetState, txHash, confirmState, confirmSteps, callToAction, errorMessage } = useMMConfirmModalState(
+    rfqTrade.trade,
```
Comment (chefjackson) [inline apps/web/src/views/Swap/MMLinkPools/components/MMCommitButtonV2.tsx:80]: same here should use `tradeToConfirm` instead if expert mode is off

**Review by chefjackson — DISMISSED** [review]

**Review by chefjackson — DISMISSED** [review]

**Review by chefjackson — APPROVED** [review]

**Pull request merged**